scripts/fetch_olean_cache.sh

```diff
@@ -20,3 +20,31 @@ fi
 [ "$new_git_sha" != "" ] || exit 0
 
 curl "$archive_url$new_git_sha.tar.gz" | tar xz src
+
+# Extracting the archive overwrites all .lean files, which is fine if we
+# downloaded an "equivalent" cache. However, since we might be using an older
+# cache, we must revert any changes made to the .lean files.
+#
+# The following commands don't make any changes to ignored files, so the .olean
+# files should be OK.
+git reset --hard HEAD
+git clean -f -d
+
+# A list of directories containing .olean files. We are being conservative to
+# avoid traversing irrelevant directories and affecting directories we do not
+# want changed (e.g. $root/.git).
+dirs="src"
+
+# Delete every <path>.olean without a matching <path>.lean.
+for olean_file in `find $dirs -name "*.olean"`
```

Comment:
I hope we never have filenames with spaces.

Reply:
Agreed. I don't think it's worth making any code changes, but I'll mention it in the comment.